splink/linker.py

```diff
@@ -813,20 +813,36 @@ def _raise_error_if_necessary_accuracy_columns_not_computed(self):
                 "Please re-run your linkage with it set to True."
             )
 
-    def initialise_settings(self, settings_dict: dict):
+    def load_settings(self, settings_dict: dict | str):
         """Initialise settings for the linker.  To be used if settings were
-        not passed to the linker on creation.
+        not passed to the linker on creation. This can either be in the form
+        of a settings dictionary or a filepath to a json file containing a
+        valid settings dictionary.
 
         Examples:
             >>> linker = DuckDBLinker(df, connection=":memory:")
             >>> linker.profile_columns(["first_name", "surname"])
             >>> linker.initialise_settings(settings_dict)
 
+            >>> linker.load_settings_from_json("my_settings.json")
+
         Args:
-            settings_dict (dict): A Splink settings dictionary
+            settings_dict (dict | str): A Splink settings dictionary or
+                the path to your settings json file.
         """
+
+        if isinstance(settings_dict, str):
```

Comment:
Can we flip this test around to be more permissive? And I don't like the .json restriction, we should just take what they supply and trust it.

```python
if not isinstance(settings_dict, dict):
    p = Path(settings_dict)  # on a separate line so the error will be clear
    settings_dict = json.loads(p.read_text())
```

Reply:
Thanks Nick, looks sensible.